tests/e2e_tests/test_tls.py

```
@@ -7,10 +7,20 @@
 from e2e_tests.utils import run_docker_compose
 
+GRPCURL_IMAGE = "fullstorydev/grpcurl"
+
 
 class TestTLS:
     """Test TLS functionality using docker-compose with mutual TLS authentication."""
 
+    @pytest.fixture(scope="class", autouse=True)
+    def grpcurl_image(self):
+        """Pre-pull the grpcurl Docker image so test timeouts only cover the actual RPC."""
+        subprocess.run(
```

Ruff (0.15.13)

[error] `subprocess` call: check for execution of untrusted input

(S603)